app/main.py

```diff
@@ -1,2 +1,151 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            keys: list[Hashable] | None = None,
+            values: list[Any] | None = None
+    ) -> None:
+        self.__capacity = 8
+        self.__size = 0
+        self.__threshold = int((2 / 3) * self.__capacity)
+        self.__table = [(None,)] * self.__capacity
+
+        if keys and values:
+            for key, value in zip(keys, values):
+                self.__setitem__(key, value)
+
+    def __repr__(self) -> str:
+        items = ", ".join(f"{key}: {value}" for key, value in self.items())
+        return f"{{{items}}}"
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        key_hash, index = self.__get_key_hash_and_index(key)
+
+        while self.__table[index][0] is not None:
+            if self.__table[index][0] == key:
+                self.__table[index] = (key, key_hash, value)
+                return
+            index += 1
+            if index == self.__capacity:
+                index = 0
+
+        self.__table[index] = (key, key_hash, value)
+        self.__size += 1
+
+        if self.__size > self.__threshold:
```

Comment:
Actually resizing takes place before a new element is added - of course if it will reach the threshold with the new element